.github/workflows/ci-e2e.yaml

```diff
@@ -67,20 +102,33 @@ jobs:
         with:
           go-version-file: go.mod
           cache: true
+      - name: Check and output changed files
+        uses: tj-actions/changed-files@7fc073d92265804a8d4e4982b637dee053daf6c5 # v42.0.7
+        id: changed-files
+        with:
+          files_ignore_yaml: |
+            doc:
+              - '**.md'
+            containers:
+              - "containers/**"
       - run: |
           swapon > swapon.txt
           sudo swapoff -a
           cat swapon.txt | tail -n+2 | awk '$2=="file" {print $1}' | sudo xargs --no-run-if-empty rm
+        if: ${{steps.changed-files.outputs.any_changed}} == 'true'
       - run: sudo mkdir /mnt/local-path-provisioner0 /mnt/local-path-provisioner1 /mnt/local-path-provisioner2
+        if: ${{steps.changed-files.outputs.any_changed}} == 'true'
       - run: make start KIND_CONFIG=kind-config_actions.yaml
         working-directory: e2e
+        if: ${{steps.changed-files.outputs.any_changed}} == 'true'
       - run: make test-upgrade
+        if: ${{steps.changed-files.outputs.any_changed}} == 'true'
         working-directory: e2e
       - run: make logs
         working-directory: e2e
-        if: always()
+        if: ${{steps.changed-files.outputs.any_changed}} == 'true' && always()
       - uses: actions/upload-artifact@v3
-        if: always()
+        if: ${{steps.changed-files.outputs.any_changed}} == 'true' && always()
         with:
           name: logs-upgrade.tar.gz
           path: e2e/logs.tar.gz
```

Comment:
How about splitting this section to a separate action file and use it ont only from this workflow but also from weekly.yaml ?
It makes job definition simpler.